R/generics.R

```diff
@@ -318,6 +318,22 @@ setMethod("obj_na_str<-", "ANY", function(obj, value) {
   obj
 })
 
+#' @rdname lab_name
+#' @export
+setGeneric("obj_align", function(obj) standardGeneric("obj_align"))
+#' @rdname lab_name
+#' @exportMethod obj_align
+setMethod("obj_align", "ANY", function(obj) attr(obj, "align", exact = TRUE))
+
+#' @rdname lab_name
+#' @export
+setGeneric("obj_align<-", function(obj, value) standardGeneric("obj_align<-"))
+#' @exportMethod obj_align<-
+#' @rdname lab_name
+setMethod("obj_align<-", "ANY", function(obj, value) {
+  attr(obj, "align") <- value
```

Comment:
the above comment about `@` is even much more true for setting. the fact that this happens to work is an implementation detail of S4 which should_never_ be relied upon.

Reply:
Thanks @gmbecker , we can rasise a new issue for the refactoring here.